### PR TITLE
Route Alias

### PR DESCRIPTION
**Pull request by lohanidamodar**

### CODE REVIEW

**Review by TorstenDittmann — COMMENTED** [review]
Are we thinking about having multiple `alias`'es?

**Comment by lohanidamodar** [comment]
> Are we thinking about having multiple `alias`'es?

If yes then we need to change the implementation. Will it be useful to have multiple aliases?

If we want to keep multiple level breaking changes then it might be useful, but don't think that's a good design choice?

**Comment by eldadfux** [comment]
> > Are we thinking about having multiple `alias`'es?
> 
> If yes then we need to change the implementation. Will it be useful to have multiple aliases?
> 
> If we want to keep multiple level breaking changes then it might be useful, but don't think that's a good design choice?

We don't need to deal with this question right now, we can always adjust the implementation.

File: src/App.php [hunk 1]
```diff
@@ -518,7 +518,7 @@ public function match(Request $request)
      * @param Route $route
      * @return self
      */
-    public function execute(Route $route, array $args = []): self
+    public function execute(Route $route, array $args = [], string $url=''): self
```
Comment (eldadfux) [inline src/App.php:521]: @lohanidamodar could you please verify how this URL format looks? especially if we have path like: `/category/:id`

Comment (lohanidamodar) [reply src/App.php:521]: Ah ok.

File: src/App.php [hunk 2]
```diff
@@ -489,11 +489,11 @@ public function match(Request $request)
             self::$routes[$method] = [];
         }
 
-        foreach (self::$routes[$method] as  $route) {
+        foreach (self::$routes[$method] as  $routeUrl => $route) {
```
Comment (eldadfux) [inline src/App.php:492]:
```suggestion
        foreach (self::$routes[$method] as $routeUrl => $route) {
```

File: src/Route.php [hunk 3]
```diff
@@ -36,6 +36,26 @@ class Route
...
+    
+    /**
+     * Alias Params
+     *
+     * @var array
+     */
+    protected $aliasParams = [];
+
+    /**
+     * is Alias route?
```
Comment (eldadfux) [inline src/Route.php:55]:
```suggestion
     * Is Alias Route?
```

**Review by eldadfux — CHANGES_REQUESTED** [review]

**Pull request merged**